extensions/wikia/CreateNewWiki/CreateNewWikiController.class.php

```diff
@@ -202,90 +206,73 @@ public function CheckDomain() {
-			// VOLDEV-10: Parse the HTML in the message
-			$this->response->setVal( self::STATUS_MSG_FIELD, wfMessage( 'cnw-error-general' )->parse() );
-			$this->response->setVal( self::STATUS_HEADER_FIELD, wfMessage( 'cnw-error-general-heading' )->escaped() );
 
-			wfProfileOut(__METHOD__);
-			return;
-		}
+		try {
+			$context = $this->getContext();
 
-		// check if user is logged in
-		if ( !$wgUser->isLoggedIn() ) {
-			$this->setUserNotLoggedInErrorResponse();
-			wfProfileOut(__METHOD__);
-			return;
-		}
+			$requestValidator = $this->getRequestValidator();
+			$request = $context->getRequest();
 
-		// check if user has confirmed e-mail
-		if ( !$wgUser->isEmailConfirmed() ) {
-			$this->setEmailNotConfirmedErrorResponse();
-			wfProfileOut(__METHOD__);
-			return;
-		}
+			$requestValidator->assertValidParams( $request );
+			$requestValidator->assertNotTorNode( $request );
 
-		// check if user is blocked
-		if ( $wgUser->isBlocked() ) {
-			$this->setUserIsBlockedErrorResponse( $wgUser );
-			wfProfileOut(__METHOD__);
-			return;
-		}
+			$userValidator = $this->getUserValidator();
+			$user = $context->getUser();
+
+			$userValidator->assertLoggedIn( $user );
+			$userValidator->assertEmailConfirmed( $user );
+			$userValidator->assertNotBlocked( $user );
+			$userValidator->assertNotExceededRateLimit( $user );
+		} catch ( ValidationException $validationException ) {
+			$context = $this->getContext();
+
+			$httpStatusCode = $validationException->getHttpStatusCode();
+
+			$headerMessageKey = $validationException->getHeaderMessageKey();
+			$headerMessageParams = $validationException->getHeaderMessageParams();
+
+			$errorMessageKey = $validationException->getErrorMessageKey();
+			$errorMessageParams = $validationException->getErrorMessageParams();
+
+			$headerMessage =
+				$context->msg( $headerMessageKey )
+					->params( $headerMessageParams )
+					->parse();
+			$errorMessage =
+				$context->msg( $errorMessageKey )
+					->params( $errorMessageParams )
+					->parse();
+
+			$this->response->setCode( $httpStatusCode );
+			$this->response->setValues( [
+				static::STATUS_FIELD => static::STATUS_ERROR,
+				static::STATUS_MSG_FIELD => $errorMessage,
+				static::STATUS_HEADER_FIELD => $headerMessage,
+			] );
 
-		// check if user is a tor node
-		if ( class_exists( 'TorBlock' ) && TorBlock::isExitNode() ) {
-			$this->setUserIsTorNodeErrorResponse();
-			wfProfileOut(__METHOD__);
 			return;
 		}
 
-		// check if user created more wikis than we allow per day
-		$numWikis = $this->countCreatedWikis($wgUser->getId());
-		if($numWikis >= self::DAILY_USER_LIMIT && $wgUser->isPingLimitable() && !$wgUser->isAllowed( 'createwikilimitsexempt' ) ) {
-			$this->setTooManyWikiCreatedByUserErrorResponse();
-			wfProfileOut(__METHOD__);
-			return;
+		$params = $this->request->getArray( 'data' );
+
+		// CE-315
+		if ( $params['wLanguage'] != self::LANG_ALL_AGES_OPT ) {
+			$params['wAllAges'] = null;
```

Comment:
Be aware of our changes related to wikis targeted to children. The checkbox is now enabled for all languages. Check with the `dev` branch please.